docs: add warning about job-intermixing (#506)
bareos/bareos: docs/manuals/source/Configuration/Director.rst

Note, you define only a single Job to do the Full, Differential, and Incremental backups since the different backup levels are tied together by a unique Job name. Normally, you will have only one Job per Client, but if a client has a really huge number of files (more than several million), you might want to split it into to Jobs each with a different FileSet covering only part of the total files.

Multiple Storage daemons are not currently supported for Jobs, if you do want to use multiple storage daemons, you will need to create a different Job and ensure that that the combination of Client and FileSet are unique.

.. warning::

  Bareos uses only :config:option:`dir/job/Client` and :config:option:`dir/job/Fileset` to determine which jobids belong together.

>   If job A and B have the same client and fileset defined, the resulting jobids will be intermixed:

Reviewer: ```suggestion
  If job A and B have the same client and fileset defined, the resulting jobids will be intermixed as follows:
```

>   When a job determines its predecessor to determine its required level and since-time, it will consider all jobs with the same client and fileset.

Reviewer: ```suggestion
  A) When a job determines its predecessor to determine its required level and since-time, it will consider all jobs with the same client and fileset.
```

Author: I'm not sure this will be formatted in a way I like, but I'll try it.

>   Likewise when restoring a client you select the fileset and all jobs that used that fileset will be considered.

Reviewer: ```suggestion
  B) When restoring a client you have to select the fileset and all jobs that used the regarding fileset will be considered.
```

>   If you want separate backups, you have to duplicate your filesets with a different name and the same content.

Reviewer: ```suggestion
  As a matter of fact, if you want separate backups, you have to duplicate your filesets with a different name and the same content.
```

> Note, you define only a single Job to do the Full, Differential, and Incremental backups since the different backup levels are tied together by a unique Job name. Normally, you will have only one Job per Client, but if a client has a really huge number of files (more than several million), you might want to split it into to Jobs each with a different FileSet covering only part of the total files.

Reviewer 2: ```suggestion
Note, you define only a single Job to do the Full, Differential, and Incremental backups since the different backup levels are tied together by a unique Job name. Normally, you will have only one Job per Client, but if a client has a really huge number of files (more than several million), you might want to split it into several Jobs each with a different FileSet covering only part of the total files.
```

> Multiple Storage daemons are not currently supported for Jobs, if you do want to use multiple storage daemons, you will need to create a different Job and ensure that that the combination of Client and FileSet are unique.

Reviewer 2: ```suggestion
Multiple Storage daemons are not currently supported for Jobs, if you do want to use multiple storage daemons, you will need to create a different Job and ensure the combination of Client and FileSet is unique.
```
